charts/spire/charts/spire-server/templates/statefulset.yaml

```diff
@@ -167,7 +167,19 @@ spec:
               subPath: policy.json
               readOnly: true
             {{ end }}
-            {{ end }}
+            {{- end }}
+            {{- with .Values.upstreamAuthority.vault }}
+            {{- if eq (.enabled | toString) "true" }}
+            {{- if eq (.k8sAuth.enabled | toString) "true" }}
+            - name: spire-psat
+              mountPath: /run/spire/secrets/kubernetes.io/serviceaccount/{{ .k8sAuth.token.name }}
+            {{- end }}
+            {{- if ne (.insecureSkipVerify | toString) "true" }}
```

Comment:
Shouldn't we just always mount this volume?

Reply:
Agreed, I will be changing this now

Reply:
I recently tested the PSAT mounting, it seems I can't mount it in default service account locations such as:
- `/run/spire/secrets/kubernetes.io/serviceaccount`
- `/var/run/spire/secrets/kubernetes.io/serviceaccount`

Would the directory of `/run/spire/secrets/kubernetes.io/psat` be sufficient?

Reply:
I've decided to change the path to `/var/run/secrets/tokens/`, which is also used by the Spire Agent and its mounted PSAT.

Let me know if you would prefer something else, otherwise feel free to resolve 👍

Reply:
> Shouldn't we just always mount this volume?

Now that I think about it, the only time we use a PSAT with the Spire Server is for attestation purposes when communicating with Vault. If we left this volume mounted, it serves no other purpose (that I currently see), which may cause confusion for operators.

What are your thoughts?